sql-cli/src/sql_cli/dag_generator.py

```diff
@@ -0,0 +1,46 @@
+from __future__ import annotations
+
+from networkx import DiGraph, depth_first_search, find_cycle, is_directed_acyclic_graph
+from sql_directory_parser import SqlFile
+
+
+class SqlFilesDAG:
+    """
+    A DAG of sql files i.e. used for finding the right order to execute the sql files in.
+
+    :param sql_files: The sql files to use for DAG generation.
+    """
+
+    def __init__(self, sql_files: set[SqlFile]) -> None:
+        self.sql_files = sql_files
+        self.nodes = [sql_file.get_variable_name() for sql_file in self.sql_files]
+        self.edges = [
+            (sql_file.get_variable_name(), parameter)
+            for sql_file in self.sql_files
+            for parameter in sql_file.get_parameters()
```

Comment:
We should only be creating edges for parameters that are of type "table"

Reply:
The "if" one line below checks that by doing this `if parameter in self.nodes  # only add edges for existing nodes`. `self.nodes` only contains file names / input tables.